main.tf

```diff
@@ -1,31 +1,112 @@
+################################################################################
+# Cluster
+################################################################################
+
 resource "aws_ecs_cluster" "this" {
-  count = var.create_ecs ? 1 : 0
+  count = var.create ? 1 : 0
+
+  name = var.cluster_name
 
-  name = var.name
+  dynamic "configuration" {
+    for_each = try([var.cluster_configuration], [])
+
+    content {
+      dynamic "execute_command_configuration" {
+        for_each = try([configuration.value.execute_command_configuration], [])
 
-  setting {
-    name  = "containerInsights"
-    value = var.container_insights ? "enabled" : "disabled"
+        content {
+          kms_key_id = try(execute_command_configuration.value.kms_key_id, null)
+          logging    = try(execute_command_configuration.value.logging, "DEFAULT")
+
+          dynamic "log_configuration" {
+            for_each = try([execute_command_configuration.value.log_configuration], [])
+
+            content {
+              cloud_watch_encryption_enabled = try(log_configuration.value.cloud_watch_encryption_enabled, null)
+              cloud_watch_log_group_name     = try(log_configuration.value.cloud_watch_log_group_name, null)
+              s3_bucket_name                 = try(log_configuration.value.s3_bucket_name, null)
+              s3_bucket_encryption_enabled   = try(log_configuration.value.s3_bucket_encryption_enabled, null)
+              s3_key_prefix                  = try(log_configuration.value.s3_key_prefix, null)
+            }
+          }
+        }
+      }
+    }
+  }
+
+  dynamic "setting" {
+    for_each = length(var.cluster_settings) > 0 ? [var.cluster_settings] : []
+
+    content {
+      name  = try(setting.value.name, "containerInsights")
```

Comment:
`containerInsights` is the wrong default value here. As soon as AWS or Terraform AWS provider adds another setting it will be a breaking change for some users. It is better to just ask users for the `name` as you already ask for the `value`. Both name and value are required according to the [documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_cluster#setting).

Reply:
ya, this isn't right to put here